.github/workflows/upload-test-specs.yml

```diff
@@ -0,0 +1,65 @@
+name: Upload AWS Device Farm test specs
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/upload-test-specs.yml
+      - examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/upload-test-specs.yml
+      - examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref_name }}-${{ github.ref_type == 'branch' && github.sha }}-${{ github.event_name == 'workflow_dispatch' }}-${{ github.event_name == 'schedule' }}
+  cancel-in-progress: true
+
+jobs:
+  upload-android-test-specs:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 15
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+          cache: pip
+
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1.7.0
+        with:
+          role-to-assume: arn:aws:iam::308535385114:role/gha_executorch_upload-frameworks-android
+          aws-region: us-east-1
+
+      - name: Only push to S3 when running the workflow manually from main branch
+        # if: ${{ github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/main' }}
```

Comment:
1. Should it be uncommented?
2. What are validated in this job on PR if the spec is only uploaded once post-merge?
3. Why uploading when running the workflow manually?

Reply:
Yeah, the testing is done, I will need to uncomment this before landing.

Reply:
Aha I see, so we don't need to trigger this on PR. `pull_request:` is added for testing purpose only.

Reply:
Is there a way to validate the test-spec on PR, instead of updating the S3 to a broken one and defer the issue discovering until the runtime. It will be nice if we could run some validation in the PR that modifies the test-spec

Reply:
Ah, we can keep `pull_request` trigger here as it will use `aws s3 cp --dry-run` which doesn't copy the file and act as a form of testing on PR

Reply:
> Is there a way to validate the test-spec on PR, instead of updating the S3 to a broken one and defer the issue discovering until the runtime. It will be nice if we could run some validation in the PR that modifies the test-spec

Good point! let me see if I can trigger the android-perf job with the new spec for testing as it's the only way to know if the spec works.

Reply:
If we don't run any validation on PR, I don't see the value of keeping the `pull_request:` other than validating the job itself runnable. 

Ideally some static analysis for the test-spec on PR would be good, just to ensure no syntax issue, mandatory things are provided in valid format, etc. 

It may not be a good idea to add dependency to the `android-perf`, for the short-term to validate the test-spec, I'm okay with it, if we can make sure we only work on a tiny model with one backend when triggering the job (defaults for on-demand workflow would be good). FYI even today we coded periodic workflow and on-demand workflow with the same defaults, later the periodic defaults will be expanded.

Reply:
I think it's ok to call `android-perf` with the small story model and the default set of parameter.  To achieve this, I make the test spec configurable as an input defaults to the S3 source of truth at https://ossci-android.s3.amazonaws.com/executorch/android-llm-device-farm-test-spec.yml